lazyscribe/experiment.py

```diff
@@ -438,6 +441,80 @@ def to_tabular(self) -> dict:
             **{("metrics", key): value for key, value in d["metrics"].items()},
         }
 
+    def promote_artifact(self, repository: Repository, name: str):
+        """Associate an artifact with a :py:class:`lazyscribe.repository.Repository`.
+
+        The purpose of this method is to move an artifact from an *ephemeral*
+        experiment to the versioned repository.
+
+        If the artifact does not exist on disk yet, this function is simply a passthrough
+        to :py:meth:`lazyscribe.repository.Repository.log_artifact`. If the artifact does
+        exist on disk already, this function will copy the artifact from the experiment
+        directory to the repository, increment the version, and call
+        :py:meth:`lazyscribe.repository.Repository.save`.
+
+        Parameters
+        ----------
+        repository : Repository
+            The :py:class:`lazyscribe.repository.Repository` to promote the artifact to.
+        name : str
+            The artifact to promote.
+
+        Raises
+        ------
+        ArtifactLogError
+            Raised if the artifact to be promoted is older than the latest version available in the
+            repository.
+        ArtifactLoadError
+            Raised if there is no artifact with the name ``name`` in the experiment.
+        """
+        for artifact in self.artifacts:
+            if artifact.name == name:
+                try:
+                    meta_ = repository.get_artifact_metadata(name)
+                    if (
+                        datetime.strptime(meta_["created_at"], "%Y-%m-%dT%H:%M:%S")
+                        > artifact.created_at
+                    ):
+                        raise ArtifactLogError(
+                            f"Artifact `{name}` is older than the latest version available in the repository."
+                        ) from None
+                    new_handler = evolve(artifact, version=meta_["version"] + 1)
+                except ValueError:
+                    new_handler = copy(artifact)
+
+                if artifact.dirty:
+                    LOG.debug(
+                        f"Artifact '{name}' does not exist on the filesystem yet."
+                    )
+                    repository.artifacts.append(new_handler)
+                else:
+                    # The artifact is on disk, we will have to copy it over
+                    curr_path = self.path / artifact.fname
+                    new_path = repository.dir / artifact.name
+                    LOG.debug(f"Copying '{curr_path!s}' to '{new_path!s}{os.sep}'")
+                    if not self.fs.isdir(f"{new_path!s}{os.sep}"):

```

Comment:
Is `os.sep` appending is required here?

Reply:
Based on `fsspec`'s documentation [here](https://filesystem-spec.readthedocs.io/en/latest/copying.html), yes. The copy was failing when I didn't include it.